src/TuxxAnimations.js

```diff
@@ -49,59 +50,323 @@ var createAnimationClass = function (transitions, customClassName) {
+            leaveActive['transition-delay'] =  delay;
+          }
+          if (enterProps) {
+            //case to accept if enterProps.delay prop is a number
+            if (typeof enterProps.delay === "number") {
+              enterProps.delay = enterProps.delay + "ms";
+            }
+            enter['transition-delay'] =  enterProps.delay;
+            enterActive['transition-delay'] = enterProps.delay;
+          }
+          if (leaveProps) {
+            //case to accept if leaveProps.delay prop is a number
+            if (typeof leaveProps.delay === "number") {
+              leaveProps.delay = leaveProps.delay + "ms";
+            }
+            leave['transition-delay'] = leaveProps.delay;
+            leaveActive['transition-delay'] = leaveProps.delay;
+          }
+
+          //vector prop work--------------------------------------------
+          //vector property takes an object consisting of origin and magnitude
+
+          //origin prop work--------------------------------------------
+          var origin = this.props.origin
+          //check if origin is an accepted origin or throw error
+          var AcceptedOrigins = ['top','right','bottom','left','center','center center','top left','top center','top right','right center','bottom right','bottom center','bottom left', 'left center']
+          if (origin) {
+            if (AcceptedOrigins.indexOf(origin) > -1) {
+              enter['transform-origin'] = origin;
+              enterActive['transform-origin'] = origin;
+              leave['transform-origin'] = origin;
+              leaveActive['transform-origin'] = origin;
+            } else {
+              throw "origin type: animation origin property is not of acceptable type. List of acceptable origins: 'center center','top left','top center','top right','right center','bottom right','bottom center','bottom left', 'left center'";
+            }
+          }
+          if (enterProps) {
+            if (AcceptedOrigins.indexOf(enterProps.vector.origin) > -1) {
+              enter['transform-origin'] = enterProps.vector.origin;
+              enterActive['transform-origin'] = enterProps.vector.origin;
+            } else {
+              throw "origin type: animation origin property is not of acceptable type. List of acceptable origins: 'center center','top left','top center','top right','right center','bottom right','bottom center','bottom left', 'left center'";
+            }
+          }
+          if (leaveProps) {
+            if (AcceptedOrigins.indexOf(leaveProps.vector.origin) > -1) {
+              leave['transform-origin'] = leaveProps.vector.origin;
+              leaveActive['transform-origin'] = leaveProps.vector.origin;
+            } else {
+              throw "origin type: animation origin property is not of acceptable type. List of acceptable origins: 'center center','top left','top center','top right','right center','bottom right','bottom center','bottom left', 'left center'";
```

Comment:
Instead of throwing errors, use invariant. Look at other files in the tuxx/src for examples.